core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -256,4 +272,33 @@ private[spark] class AppClient(
       endpoint = null
     }
   }
+
+  /**
+   * Request executors from the Master by specifying the total number desired,
+   * including existing pending and running executors.
+   *
+   * @return whether the request is acknowledged.
+   */
+  def requestTotalExecutors(requestedTotal: Int): Boolean = {
```

Comment:
is it necessary to validate the value of `requestedTotal`, like `>= 0`? though negative numbers does not bring any impact on the correctness of the program (if I understand code correctly)